addons/project/models/project.py

```diff
@@ -1243,11 +1384,14 @@ def write(self, vals):
             recurrence_domain = []
             if recurrence_update == 'subsequent':
                 for task in self:
-                    recurrence_domain = OR([recurrence_domain, ['&', ('recurrence_id', '=', task.recurrence_id.id), ('create_date', '>=', task.create_date)]])
+                    recurrence_domain = expression.OR([recurrence_domain, ['&', ('recurrence_id', '=', task.recurrence_id.id), ('create_date', '>=', task.create_date)]])
             else:
                 recurrence_domain = [('recurrence_id', 'in', self.recurrence_id.ids)]
             tasks |= self.env['project.task'].search(recurrence_domain)
 
+        if portal_can_create:
+            tasks = tasks.sudo()
```

Comment:
same question: why do we need `sudo()` if we've determined that the user can `write()`?

Reply:
@odony not sure whether it was added after your comment, but there's now a comment saying it's because of `rating.rating` being updated afterwards.